### ISSUE TITLE
MySQL - On Update not taking expression

### ISSUE THREAD

**Issue by Sibz** [apache/datafusion-sqlparser-rs#543]
```sql 
CREATE TABLE foo (`modification_time` DATETIME ON UPDATE CURRENT_TIMESTAMP())
```
does not work

**Comment by Sibz** [comment]
I am trying to resolve this 
https://github.com/rustanigans/sqlparser-rs/tree/fix-mysql-on-update

But unable to work out how to make the test:
https://github.com/rustanigans/sqlparser-rs/blob/2585a03b6d7e700e0fe619e53e641af05c86bf9a/tests/sqlparser_mysql.rs#L853-L875

Also the ColumnOption::DialectSpecific only takes vec of tokens, so I had to add one for this that takes an expr:
https://github.com/rustanigans/sqlparser-rs/blob/2585a03b6d7e700e0fe619e53e641af05c86bf9a/src/ast/ddl.rs#L378-L379

If anyone could provide a bit of guidance I will make a PR once resolved.

**Comment by alamb** [comment]
I think if you make a PR that is likely the best way to get guidance.

**Comment by AugustoFKL** [comment]
@Sibz looking at MySQL's `CREATE TABLE` doc ([1](https://dev.mysql.com/doc/refman/8.0/en/create-table.html)), I couldn't find why you're trying to give it an expression, as MySQL seems to follow the standard when it comes to the `ON UPDATE` clause([2](https://jakewheat.github.io/sql-overview/sql-2016-foundation-grammar.html#update-rule)).

You're trying to add the `ON UPDATE CURRENT_TIMESTAMP` clause ([3](https://dev.mysql.com/doc/refman/8.0/en/timestamp-initialization.html)), which is a unique kind of parameter.

[1] : https://dev.mysql.com/doc/refman/8.0/en/create-table.html
[2] : https://jakewheat.github.io/sql-overview/sql-2016-foundation-grammar.html#update-rule

**Comment by Sibz** [comment]
Yes your right, it doesn't take an expression per se, only takes `CURRENT_TIMESTAMP` or `NOW` but something must be provided, currently it's broken as will not parse those key words.

**Comment by Sibz** [comment]
And further, we must support all synonyms of CURRENT_TIMESTAMP: 
`CURRENT_TIMESTAMP()`, `NOW()`, `LOCALTIME`, `LOCALTIME()`, `LOCALTIMESTAMP` and `LOCALTIMESTAMP()`
Parse considers the ones without parenthesis keywords but the ones with as expressions.

I will review my code and test as soon as I have chance, though I am very busy these days.

**Comment by LucaCappelletti94** [comment]
Completed by #685.
